src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.test.tsx

```diff
@@ -0,0 +1,191 @@
+    value: 'three',
+    label: 'three',
+    isSelected: false
+  },
+  {
+    value: 'four',
+    label: 'four',
+    isSelected: false
+  },
+  {
+    value: 'five',
+    label: 'five',
+    isSelected: false
+  }
+];
+describe('<CheckboxWithSelects />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  let wrapper: any;
+  const defaultProps = {
+    onChange: onChange,
+    label: 'foo',
+    selectedOptions: [],
+    selectOptions: selectOptions
+  };
+
+  it('renders without error', () => {
+    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+    expect(wrapper.find(CheckboxWithSelects).length).toEqual(1);
+  });
+
+  it('does not check the checkbox when there are no options selected', () => {
+    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+
+    expect(wrapper.find(Checkbox).prop('checked')).toBe(false);
+  });
+
+  it('does not display any Select when the checkbox is unchecked', () => {
+    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+
+    expect(wrapper.find(Checkbox).prop('checked')).toBe(false);
+    expect(wrapper.find(Select).length).toBe(0);
+  });
+
+  it('displays one Select when the checkbox is checked', () => {
+    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+
+    wrapper
+      .find(Checkbox)
+      .find('.defaultCheckbox')
+      .simulate('click');
+    expect(wrapper.find(Select).length).toBe(1);
+  });
+
+  it('automatically checks the checkbox if atleast one option is selected', () => {
+    wrapper = mount(
+      <CheckboxWithSelects {...defaultProps} selectedOptions={['one']} />
+    );
+
+    expect(wrapper.find(Checkbox).prop('checked')).toBe(true);
+  });
+
+  it('does not display the remove button next to the Select if no option is selected ', () => {
+    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+
+    wrapper
+      .find(Checkbox)
+      .find('.defaultCheckbox')
+      .simulate('click');
+    expect(wrapper.find(Select).length).toBe(1);
+    expect(wrapper.find('.removeIconHolder').length).toBe(0);
+  });
+
+  it('displays the remove button next to the Select if an option is selected', () => {
+    wrapper = mount(
+      <CheckboxWithSelects {...defaultProps} selectedOptions={['one']} />
+    );
+
+    expect(wrapper.find('.removeIconHolder').length).toBe(1);
+  });
+
+  it('displays the Plus button when one option is selected', async () => {
+    wrapper = mount(
+      <CheckboxWithSelects {...defaultProps} selectedOptions={['one']} />
+    );
+
+    expect(wrapper.find('.addIconHolder').length).toBe(1);
+  });
+
+  it('displays another select when the plus button is clicked', () => {
+    wrapper = mount(
+      <CheckboxWithSelects {...defaultProps} selectedOptions={['one']} />
+    );
+
+    wrapper
+      .find('.addIconHolder')
+      .find(ImageButton)
+      .simulate('click');
+    expect(
+      wrapper
+        .find(Select)
+        .last()
+        .prop('options')
+    ).toHaveLength(4);
```

Comment:
The description of this test talks about displaying another Select element; while the test checks for the number of options in the select element. Either the description is incorrect, or the test is.

Reply:
Updated it